web/packages/teleport/src/Discover/ConnectMyComputer/TestConnection/TestConnection.tsx

```diff
@@ -68,168 +84,227 @@ export const TestConnection = (props: AgentStepProps) => {
     )
   );
 
+  const fetchLoginsAndUpdateLoginSelection = async (signal: AbortSignal) => {
+    // fetchLogins is from useAsync which always resolves the underlying promise and uses a Go-style
+    // API for error handling.
+    const [logins, err] = await fetchLogins(signal);
+    if (err) {
+      return;
+    }
+
+    if (logins.length > 0) {
+      setSelectedLoginOpt(mapLoginToSelectOption(logins[0]));
+
+      // Start the test automatically if there are no logins to choose from.
+      if (logins.length == 1) {
+        const { mfaRequired } = await testConnection(logins[0]);
+
+        // If MFA is required, let's just wait for the user to start the connection themselves.
+        if (mfaRequired) {
+          cancelMfaDialog();
+        }
+      }
+    }
+  };
+
   useEffect(() => {
     abortController.current = new AbortController();
 
     if (fetchLoginsAttempt.status === '') {
-      fetchLogins(abortController.current.signal);
+      fetchLoginsAndUpdateLoginSelection(abortController.current.signal);
     }
 
     return () => {
       abortController.current.abort();
     };
   }, []);
 
+  function startSshSession(login: string) {
+    const url = cfg.getSshConnectRoute({
+      clusterId: node.clusterId,
+      serverId: node.id,
+      login,
+    });
+
+    openNewTab(url);
+  }
+
+  function testConnection(login: string, mfaResponse?: MfaAuthnResponse) {
+    return runConnectionDiagnostic(
+      {
+        resourceKind: 'node',
+        resourceName: props.agentMeta.resourceName,
+        sshPrincipal: login,
+      },
+      mfaResponse
+    );
+  }
```

Comment:
In #34952 I mentioned sharing these two functions between Connect My Computer and Server connection tests, but I think for now I'm going to keep them this way. It's not much code, in the end it calls functions from `useConnectionDiagnostic` which is already shared. And to make the diagnostic messages be more relevant to Connect My Computer, I'll have to add some extra parameter to `runConnectionDiagnostic` anyway.

So in the end I don't think there's much to be gained from sharing these two functions.